runtime/onert/api/python/package/infer/session.py

```diff
@@ -134,6 +108,30 @@ def infer(
         # Return the output buffers.
         return (self.outputs.copy(), metrics) if measure else self.outputs.copy()
 
+    def _update_inputs_tensorinfo(self, new_infos: List[tensorinfo]) -> None:
+        """
+        Update all input tensors' tensorinfo at once.
+
+        Args:
+            new_infos (list[tensorinfo]): A list of updated tensorinfo objects for the inputs.
+
+        Raises:
+            ValueError: If the number of new_infos does not match the session's input size,
+                        or if any tensorinfo contains a negative dimension.
+        """
+        num_inputs: int = self.session.input_size()
+        if len(new_infos) != num_inputs:
+            raise ValueError(
+                f"Expected {num_inputs} input tensorinfo(s), but got {len(new_infos)}.")
+
+        for i, info in enumerate(new_infos):
+            # Check for any negative dimension in the specified rank
+            if any(d < 0 for d in info.dims[:info.rank]):
+                raise ValueError(
+                    f"Input tensorinfo at index {i} contains negative dimension(s): "
+                    f"{info.dims[:info.rank]}")
+            self.session.set_input_tensorinfo(i, info)
+
     @contextmanager
     def _time_block(self, metrics: Dict[str, float], key: str, mesure: bool):
```

Comment:
(optional) not related this PR, typo here.

Reply:
OK, I will fix it on the next PR.